org.lflang/src/org/lflang/generator/c/TypeParameterizedReactor.java

```diff
@@ -76,9 +90,23 @@ public InferredType resolveType(InferredType t) {
     return InferredType.fromAST(resolveType(t.astType));
   }
 
+  public String uniqueName() {
+    String name = reactor.getName().toLowerCase();
+    if (uniqueNames.containsKey(this)) return uniqueNames.get(this);
+    if (nameCounts.containsKey(name)) {
+      int currentCount = nameCounts.get(name);
+      nameCounts.put(name, currentCount + 1);
```

Comment:
I noticed that there is one shared counter for all names. Not that it matters, but would it be worth it to have a counter per name for the same of readability?

Reply:
But I think I do use different counters for different names. `currentCount` is retrieved from a hash table the keys of which are the names.

Reply:
Oh, you're right. I missed that. Great!